main_app/views.py

```diff
@@ -108,10 +143,36 @@ def items_details(request, item_id):
     categories_item_doesnt_have = Category.objects.exclude(id__in=id_list)
     return render(request, 'items/detail.html', {'item':item, 'categories':categories_item_doesnt_have})
 
+# association stuff
 def assoc_category(request, item_id, category_id):
-    Item.objects.get(id=item_id).categories.add(category_id)
+    item = get_object_or_404(Item, id=item_id)
```

Comment:
Item.objects.get(id=item_id).categories.add(category_id) has been replaced here for error handling purposes. Just in case we end up stuff up the code somewhere in future, we'll get an error message instead of DoesNotExist